### PR TITLE
Validate python and dbnd version between local and spark envs

### PR DESCRIPTION
**Pull request by turbaszek**
Apart from this change I tested pickling with:
- dump 3.8 > load 3.6
- dump 3.6 > load  3.8
- dump 3.8 > load  3.7
- dump 3.6 > load 3.6

I used a dictionary (nothing special) and in all cases pickling worked as expected.

### CODE REVIEW

File: modules/dbnd/src/dbnd/_core/task/task.py [hunk 1]
```diff
@@ -386,6 +386,6 @@ def dbnd_run(self):
     [
         v
         for k, v in Task.__dict__.items()
-        if isinstance(v, ParameterDefinition) and v.value_type_str is not "Target"
+        if isinstance(v, ParameterDefinition) and v.value_type_str != "Target"
```
Comment (turbaszek) [inline modules/dbnd/src/dbnd/_core/task/task.py:389]: Style issue, rises a warning in Py3.8

File: plugins/dbnd-gcp/src/dbnd_gcp/dataproc/dataproc.py [hunk 2]
```diff
@@ -1,7 +1,6 @@
 import logging
 
 from airflow.contrib.hooks.gcp_dataproc_hook import DataProcHook
-from airflow.contrib.hooks.gcs_hook import GoogleCloudStorageHook
```
Comment (turbaszek) [inline plugins/dbnd-gcp/src/dbnd_gcp/dataproc/dataproc.py:4]: Unused import

File: modules/dbnd/src/dbnd/_core/cli/cmd_execute.py [hunk 3]
```diff
@@ -7,13 +7,51 @@
...
+        spark_dbnd_version = get_dbnd_version()
+        if expected_python_version != spark_python_version:
+            raise DbndVersionsClashError(
+                "You submmited job using Python {} but the Spark cluster uses Python {}. To "
+                "assure execution consistency use the same version in both places.".format(
+                    expected_python_version, spark_python_version
+                )
+            )
+        if expected_dbnd_version != spark_dbnd_version:
+            raise DbndVersionsClashError(
```
Comment (jonathanshir) [inline modules/dbnd/src/dbnd/_core/cli/cmd_execute.py:48]: Why raise? I think we're better off warning the user, and if code works - then so be it. WDYT?

Comment (turbaszek) [reply modules/dbnd/src/dbnd/_core/cli/cmd_execute.py:48]: My main worry is the following situation:
- user submits the job
- something crashes miserably
- the error is due to databand version mismatch
- user misses a warning
- user reaches to dbnd for support

If we know that we are backward / forward compatible within major version then we should only check if the major is the same. As I'm not that familiar with what users tend to do, I'm happy to do what you suggest as best approach.

Comment (jonathanshir) [reply modules/dbnd/src/dbnd/_core/cli/cmd_execute.py:48]: I think that we can't take this decision for the user. We do our best -- write warning, a lot of text so its noticeable, but keep trying to run (best effort). I would simply write this in the message.
"You submitted job using version ... To assure execution consistency use the same version in both places. Continuing execution in best effort...
Users will notice, we differentiate spark engine logs and our bootstrap logs, I think this is our best approach, so reduce to warning 💯

**Pull request merged**